internal/nginx/config/http/config.go

```diff
@@ -5,18 +5,26 @@ type Server struct {
 	SSL           *SSL
 	ServerName    string
 	Locations     []Location
+	Maps          []Map
```

Comment:
Since maps do not belong to server - they are part of the `http` contexts - I suggest we store them differently. For example:
```
type Config struct {
  Maps[] Map
 Servers []Server
}
```
this way the problem of duplicating maps will become more visible - since we will store all maps in a single slice.


Thoughts?

Reply:
Absolutely agreed - I have changed the location of where the maps logic occurs, created a separate maps template, and deduplicated the headerMaps across all servers. Let me know what you think!